src/internet_identity/src/storage/tests.rs

```diff
@@ -80,6 +82,112 @@ fn should_recover_header_from_memory_v7() {
     assert_eq!(storage.version(), 7);
 }
 
+fn add_test_anchor_data<M: Memory + Clone>(storage: &mut Storage<M>, number_of_anchors: usize) {
+    for _ in 0..number_of_anchors {
+        let (anchor_number, mut anchor) = storage
+            .allocate_anchor()
+            .expect("Failure allocating an anchor.");
+        anchor
+            .add_device(sample_device())
+            .expect("Failure adding a device");
+        storage.write(anchor_number, anchor.clone()).unwrap();
+    }
+}
+
+fn range_as_hex(memory: &RefCell<Vec<u8>>, offset: u64, length: usize) -> String {
+    let mut buf = vec![0u8; length];
+    memory.read(offset, &mut buf);
+    let s = hex::encode(buf);
+    let mut answer = String::new();
+    for (i, c) in s.chars().enumerate() {
```

Comment:
Hm, I stumbled on this loop because it was not immediately obvious to me that you also do formatting here.
Either add a comment or drop the formatting entirely.

Reply:
Added a comment (with the formatting it is easier to debug).